File: component/webdav/src/main/java/org/exoplatform/services/webdav/common/property/WebDavProperty.java
/**
* Copyright 2001-2006 The eXo Platform SARL         All rights reserved.  *
* Please look at license.txt in info directory for more license detail.   *
*/

package org.exoplatform.services.webdav.common.property;

import org.exoplatform.services.webdav.common.document.XmlSerializable;
import org.exoplatform.services.webdav.common.property.factory.MappingTable;
import org.exoplatform.services.webdav.common.property.factory.PropertyConfigTable;
import org.exoplatform.services.webdav.common.resource.WebDavResource;
import org.exoplatform.services.webdav.common.response.Href;

/**
 * Created by The eXo Platform SARL
 * Author : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 * @version $Id: WebDavProperty.java 12525 2007-02-02 12:26:47Z gavrikvetal $
 */

public interface WebDavProperty extends XmlSerializable {
  
  void setConfiguration(MappingTable mappingTable, PropertyConfigTable propertyConfigTable);

  void setStatus(int status);
  
  int getStatus();
  
  void setName(String propertyName);
  
  String getName();
    
  void setValue(String propertyValue);
  
  String getValue();
  
  void setIsMultiValue();
  
  boolean isMultiValue();
  
  void addMultiValue(String propertyValue);

  boolean refresh(WebDavResource resource, Href href);
  
  boolean set(WebDavResource resource);
  
  boolean remove(WebDavResource resource);
  
}

File: applications/dotnet/exo_jcr.webdav.csclient/DavProperties/WebDavProperty.cs
/***************************************************************************
 * Copyright 2001-2007 The eXo Platform SAS          All rights reserved.  *
 * Please look at license.txt in info directory for more license detail.   *
 **************************************************************************/
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using System.Collections;

/**
 * Created by The eXo Platform SARL
 * Authors : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 *         : Max Shaposhnik <uy7c@yahoo.com>
 * @version $Id:
 */

namespace exo_jcr.webdav.csclient.DavProperties
{
    public class WebDavProperty
    {

        private String propertyName;

        private String propertyValue;

        private bool _isMultivalue = false;
        private ArrayList propertyValues;

        private int status = DavStatus.NOT_FOUND;

        public WebDavProperty(String propertyName)
        {
            this.propertyName = propertyName;
        }

        public void setValue(String propertyValue)
        {
            if (_isMultivalue)
            {
                propertyValues.Add(propertyValue);
            }
            else
            {
                this.propertyValue = propertyValue;
            }            
        }

        public void setIsMultivalue()
        {
            _isMultivalue = true;
            propertyValues = new ArrayList();
            propertyValues.Add(propertyValue);
        }

        public bool isMultivalue()
        {
            return _isMultivalue;
        }

        public virtual void init(XmlTextReader reader)
        {
            if (reader.IsEmptyElement) {
                return;
            }
            while (reader.Read())
            {

                switch (reader.NodeType)
                {
                    case XmlNodeType.Element:

                        break;

                    case XmlNodeType.Text:
                        setValue(reader.Value);
                        break;

                    case XmlNodeType.EndElement:
                        if (reader.Name.EndsWith(propertyName))
                        {
                            return;
                        }
                        throw new XmlException("Malformed response at line " + reader.LineNumber + ":" + reader.LinePosition, null);                                                
                }

            }
        }

        private void parseElement(XmlTextReader reader)
        {
            if (reader.IsEmptyElement)
            {
                return;
            }
            while (reader.Read())
            {

                switch (reader.NodeType)
                {
                    case XmlNodeType.Text:
                        setValue(reader.Value);
                        break;

                    case XmlNodeType.EndElement:
                        if (reader.Name.EndsWith(propertyName))
                        {
                            return;
                        }
                        throw new XmlException("Malformed response at line " + reader.LineNumber + ":" + reader.LinePosition, null);
                }

            }

        }

        public String getPropertyName()
        {
            return propertyName;
        }

        public void setStatus(int status)
        {
            this.status = status;
        }

        public int getStatus()
        {
            return status;
        }

        public String getTextContent()
        {
            return propertyValue;
        }

        public ArrayList getValues()
        {
            return propertyValues;
        }

    }
}
